PhaseEstimation/Tests.qs

```diff
@@ -100,7 +100,7 @@ namespace Quantum.Kata.PhaseEstimation {
         EqualityFactI(nu, 1, $"You are allowed to call Controlled U exactly once, and you called it {nu} times");
     }
 
-    @Test("Microsoft.Quantum.Katas.CounterSimulator")
+    @Test("QuantumSimulator")
```

Comment:
I'm not sure this test should be switched to `QuantumSimulator`. It calls `Test1BitPEOnOnePair` which uses `GetOracleCallsCount` which is implemented only in `CounterSimulator`. Have you checked that the test still does the counting correctly, i.e., fails if the solution calls too many `Controlled U`s?

Reply:
I got the same doubt initially. But when I run it with the reference solution the test passes.
When I add more `Controlled U` I get the expected error message:
```
You are allowed to call Controlled U exactly once, and you called it 3 times
	Expected:	1
	Actual:	3
Try again!
```

But to be sure I can switch this back to `CounterSimulator`

Reply:
I dug deeper into this, and it turns out that IQ# (and notebooks) always use `CounterSimulator` (see https://github.com/microsoft/QuantumKatas/issues/580), but running the tests from VS/VS Code/command line will expose that failure (with a very cryptic error message, for which I'll file an issue in a moment). So please change the tests that have the operations from `CounterSimulator` back to use `CounterSimulator`, and double-check that the tests pass with the reference implementations for all tasks when executed outside of notebooks.